### PR TITLE
fix: result computation

### PR DESCRIPTION
**Pull request by Mohamed-Hacene**
The framework builder UI lets you tag each question choice with a compliance result (compliant, non_compliant, partially_compliant, not_applicable), but the requirement-assessment recomputation was collapsing every non-empty string to a boolean. As a result, picking "Non-compliant" on a question still produced a Compliant requirement assessment.

This PR introduces a proper semantic resolver + worst-wins aggregation, shared by backend and frontend, while staying backward-compatible with legacy true / false literals shipped in existing library YAMLs.

## Summary by CodeRabbit

* **New Features**
  - Per-choice semantic compliance values (compliant, non_compliant, partially_compliant, not_applicable); canonicalization, worst‑wins aggregation, weighted score calc (sum/mean) with clamping; live computed requirement result badges and flash‑mode live computation.

* **Bug Fixes**
  - Preserve manual requirement results when no choices contribute; ignore/warn on unknown compute_result tokens; stricter dependency visibility.

* **Documentation**
  - Expanded authoring, installation, product and migration/recompute guidance; updated examples.

* **Tests**
  - Extensive coverage for resolution, aggregation, legacy mappings, multi‑choice and edge cases.

* **Chores**
  - Bulk recompute command (dry‑run, batching, post‑update hooks); test tooling mocks.

### CODE REVIEW

File: backend/core/utils.py [hunk 1]
```diff
@@ -35,10 +35,76 @@ def extract_node_id(urn: str | None) -> str | None:
...
+    Supports both legacy boolean literals ("true"/"false"/"1"/"0") and the
+    semantic values produced by the framework builder UI ("compliant",
+    "non_compliant", "partially_compliant", "not_applicable").
+
+    Returns None when the choice should not contribute to the overall result
+    (i.e. compute_result is None or an empty string).
+    """
+    if compute_result is None:
+        return None
+    value = compute_result.strip().lower()
```
[the remark this answers is not in the document]
Comment (Mohamed-Hacene) [reply backend/core/utils.py:95]: The type hint is str | None and QuestionChoice.compute_result is a Django CharField, so the only real call sites already feed strings (the YAML loader converts true/false bools to "true"/"false" before .save() — covered by test_import_compute_result_bool_to_str). Adding isinstance(bool/int) branches would be defensive coding for cases that don't occur in this codebase; if a non-string ever reaches the resolver, a TypeError is the right signal rather than silent coercion.

File: backend/scripts/convert_library_v2.py [hunk 2]
```diff
@@ -809,29 +809,43 @@ def _handle_framework(obj, library, object_blocks, prefix_to_urn, compat_mode, v
...
+                        elif v in SEMANTIC_COMPUTE_RESULT_VALUES:
+                            choices[i]["compute_result"] = v
+                        else:
                             raise ValueError(
                                 f"(answers_definition) Invalid compute_result value '{val}' "
-                                f"for answer ID '{answer_id}', choice #{i + 1}. Must be 'true', 'false', '/' (= 'undefined') or empty."
+                                f"for answer ID '{answer_id}', choice #{i + 1}. Must be one of "
+                                f"'compliant', 'non_compliant', 'partially_compliant', "
+                                f"'not_applicable', 'true' (= 'compliant'), 'false' (= 'non_compliant'), "
+                                f"'/' (= 'undefined'), or empty."
```
[the remark this answers is not in the document]
Comment (Mohamed-Hacene) [reply backend/scripts/convert_library_v2.py:846]: Excel 1/0 cells come back from openpyxl as int and crash at val.lower() before this check, so the converter has never accepted them. The runtime resolver tolerates "1"/"0" only for historical string-coerced data, the Excel surface intentionally documents just true / false / / / empty + the semantic values, and adding "1"/"0" would advertise a path the converter doesn't really support.

File: backend/core/migrations/0170_backfill_requirement_assessment_results.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+"""
+
+from django.db import migrations
+
+BATCH_SIZE = 500
+
+
+def backfill_results(apps, schema_editor):
+    from core.models import RequirementAssessment
+
```
[the remark this answers is not in the document]
Comment (nas-tabchiche) [reply backend/core/migrations/0170_backfill_requirement_assessment_results.py]: This must be addressed

File: backend/core/migrations/0170_backfill_requirement_assessment_results.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    batch = []
+    for ra in queryset.iterator(chunk_size=BATCH_SIZE):
+        ra.recompute_assessment()
+        batch.append(ra)
+        if len(batch) >= BATCH_SIZE:
+            RequirementAssessment.objects.bulk_update(batch, fields)
+            batch = []
+    if batch:
+        RequirementAssessment.objects.bulk_update(batch, fields)
```
[the remark this answers is not in the document]
Comment (nas-tabchiche) [reply backend/core/migrations/0170_backfill_requirement_assessment_results.py]: We do not have a multi-DB setup at the moment. We are still inconsistent on db aliases across migration files, but this can definitely be deferred.

File: backend/core/migrations/0170_backfill_requirement_assessment_results.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mohamed-Hacene) [reply backend/core/migrations/0170_backfill_requirement_assessment_results.py]: Addressed. The migration now uses apps.get_model() and inlines the recompute logic with raw enum strings, importing only the four stateless helpers from core.utils (_build_answer_context, _is_question_visible, resolve_compute_result, aggregate_compute_results). No model class is imported from core.models.

File: backend/core/migrations/0170_backfill_requirement_assessment_results.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Mohamed-Hacene) [reply backend/core/migrations/0170_backfill_requirement_assessment_results.py]: Done. db_alias is now used on the read queryset and on both bulk_update calls. Related-manager reads inside the loop (ra.requirement.questions, ra.answers) inherit the alias automatically because each ra is loaded via .using(db_alias) and Django propagates _state.db to related lookups.

**Review by eric-intuitem — DISMISSED** [review]
LGTM

**Comment by Mohamed-Hacene** [comment]
@coderabbit review

**Comment by Mohamed-Hacene** [comment]
Nice catch on the migration @eric-intuitem, I'm putting the PR back into draft as I've spotted some bad behavior

**Comment by Mohamed-Hacene** [comment]
@coderabbitai review

File: backend/core/management/commands/recompute_assessment_results.py [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+BATCH_SIZE = 500
+RESULT_UPDATE_FIELDS = ["score", "result", "is_scored", "updated_at"]
+
+
+class _NullContext:
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc, tb):
+        return False
```
Comment (monsieurswag) [inline backend/core/management/commands/recompute_assessment_results.py]: You can use `contextlib.nullcontext` instead.

**Review by nas-tabchiche — CHANGES_REQUESTED** [review]
Functional OK. A few things to change before we can merge:
- Docs use `poetry`, we just migrated to `uv`
- The `recompute_assessments_results` management command is 200+ lines long and untested. We need at least a minimal test harness for this one even though this is a one-off.

**Review by nas-tabchiche — APPROVED** [review]
LGTM
